platform-hub-api/spec/models/kubernetes_cluster_spec.rb

```diff
@@ -2,6 +2,70 @@
 
 RSpec.describe KubernetesCluster, type: :model do
 
+  describe 'scopes' do
+
+    describe 'by_alias' do
+      let(:value) { 'foo' }
+
+      subject { KubernetesCluster.by_alias(value) }
+
+      context 'when no aliases are set' do
+        let!(:cluster) { create :kubernetes_cluster, aliases: [] }
+
+        it 'should not find any clusters' do
+          expect(subject.count).to be 0
+        end
+      end
+
+      context 'when the value is set in an aliases list' do
+        let!(:cluster) { create :kubernetes_cluster, aliases: [ value, 'another' ] }
+
+        it 'should find the cluster' do
+          expect(subject.entries).to eq [ cluster ]
+        end
+      end
+
+      context 'when the value is not used' do
+        let!(:cluster) { create :kubernetes_cluster, aliases: [ 'another' ] }
+
+        it 'should not find any clusters' do
+          expect(subject.count).to be 0
+        end
+      end
+    end
+
+    describe 'by_name_or_alias' do
+      let(:value) { 'foo' }
+
+      subject { KubernetesCluster.by_name_or_alias(value) }
```

Comment:
Super nitpick here: Could possibly include an extra test case for `by_name_or_alias` where no name is defined. You have a similar test case for `by_alias` where no name or alias is provided.

Reply:
I do like your thinking here – only thing is `name` is actually a required field, so will always have a value.

Unless I've misunderstood what you meant?

Reply:
Aha okay that's fine, I was just thrown off by the earlier tests that do: `let!(:cluster) { create :kubernetes_cluster, aliases: [] }` (no name defined)

Reply:
Ahhh yeah, we're using "factories" in our tests to help build/create pre-canned objects that have necessary bits already set up (and then you can override as you please). Check out https://github.com/UKHomeOffice/platform-hub/blob/master/platform-hub-api/spec/factories/kubernetes_clusters.rb for the `KubernetesCluster` factory.

The really cool thing about this is you don't need to do lots of object/model initialisation in many different places :)